### PR TITLE
Fix warning situation: UserWarning: max_length is ignored when padding=True"

### LINKED ISSUE #13826: Tokenizer - Raises wrong "UserWarning: `max_length` is ignored when `padding`=`True`"
In the newest version of transformers (4.11.2 & 4.12.0.dev0) I get the following warning:
```
C:\Anaconda3\envs\sbert\lib\site-packages\transformers\tokenization_utils_base.py:2227: UserWarning: `max_length` is ignored when `padding`=`True`.
  warnings.warn("`max_length` is ignored when `padding`=`True`.")
```

Code to re-produce:
```python
from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained('bert-base-uncased')
texts = ["Short sentence", "A really really really really really long sentence to test max length"]

output = tokenizer(texts, padding=True, truncation=True, max_length=5, return_tensors='pt')
print(output['input_ids'].shape)

output = tokenizer(texts, padding=True, truncation=True, return_tensors='pt')
print(output['input_ids'].shape)
```

Output:
```
C:\Anaconda3\envs\sbert\lib\site-packages\transformers\tokenization_utils_base.py:2227: UserWarning: `max_length` is ignored when `padding`=`True`.
  warnings.warn("`max_length` is ignored when `padding`=`True`.")
torch.Size([2, 5])
torch.Size([2, 14])
```` 

As we see, max_length is not ignored when padding = True. It truncates the text as expected to a max_length of 5.

I would say that the warning is incorrect and should not be raised. 

Should I fix it?

Or is it really intended that max_length is ignored when padding=True? This would be horrible, I want to truncate my text to a certain max_length.

### PR DESCRIPTION
**Pull request by shirayu**
# What does this PR do?

Fixes #13826

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@sgugger @LysandreJik

### CODE REVIEW

**Comment by nreimers** [comment]
The change looks good to me.

**Review by sgugger — COMMENTED** [review]
I don't think the warning should be just completely removed as it has some importance. We could check the `truncation` argument` and leave the warning if it's not set to `True`.

**Comment by nreimers** [comment]
> I don't think the warning should be just completely removed as it has some importance. We could check the `truncation` argument`and leave the warning if it's not set to`True`.

You mean `truncation=False` and `max_length=10` => Should that not raise another warning, that `max_length` is ignored when you have set `truncation` to `False`.

I would argue that `max_length` and `padding=True` has no connection => hence, create no warning

**Comment by sgugger** [comment]
I don't think users are confused by what is happening when `truncation=False`, they are confused by what happens when `padding=True`. Remember that most users don't read the doc so they don't know `padding=True` means "longest" and not "max length".

**Comment by nreimers** [comment]
> I don't think users are confused by what is happening when `truncation=False`, they are confused by what happens when `padding=True`. Remember that most users don't read the doc so they don't know `padding=True` means "longest" and not "max length".

Yes, that makes sense.

So this warning can be raised when `truncation == False or truncation == 'do_not_truncate'`

**Comment by shirayu** [comment]
Thank you for your comments.
I updated the code and title.

File: src/transformers/tokenization_utils_base.py [hunk 1]
```diff
@@ -2223,8 +2223,10 @@ def _get_padding_truncation_strategies(
         elif padding is not False:
             if padding is True:
                 if verbose:
-                    if max_length is not None:
-                        warnings.warn("`max_length` is ignored when `padding`=`True`.")
+                    if max_length is not None and (truncation is False or truncation == "do_not_truncate"):
+                        warnings.warn(
+                            "`max_length` is ignored when `padding`=`True` and `truncation` is False or `do_not_truncate`."
```
Comment (sgugger) [inline src/transformers/tokenization_utils_base.py:2228]:
```suggestion
                            "`max_length` is ignored when `padding`=`True` and there is no truncation strategy. "
                            "To pad to max length, use `padding='max_length'`."
```

**Review by sgugger — APPROVED** [review]
Thanks for adapting, I just have a tweak to the warning message.

**Review by sgugger — APPROVED** [review]

**Pull request merged**